File: programers/level2/Feature_Development.go
func solution(progresses []int, speeds []int) []int {
    
    var max, count int = 0, 0;
    result := []int{};
    
    for idx, n := range progresses {
        var remainPeriod int = 100 - n;
        var completPeriod int = 0;
        
        if ( remainPeriod % speeds[idx] != 0 ) {
            completPeriod = (remainPeriod / speeds[idx]) + 1;
        } else {
            completPeriod = remainPeriod / speeds[idx];
        }
        
        if( max < completPeriod ){
            result = append(result, count);
            count = 1;
            max = completPeriod;
        } else {
            count++;
        }
    }

    result = append(result[:0], result[1:]...);
    result = append(result, count);
    return result
}

File: programers/level2/Feature_Development.js
function solution(progresses, speeds) {
    let [count, max] = [0, 0];
    
    return (progresses.reduce((prev, curr, idx) => {
        const timeRemain = 100 - curr;
        const day = parseInt(timeRemain / speeds[idx]);
        const completDay = timeRemain % speeds[idx] ? day + 1 : day;
        
        if(max < completDay){
            prev.push(count);
            max = completDay;
            count = 1;
        } else {
            count++;
        }
        
        return prev;
    }, []).filter(count => count)).concat(count);
}